impl/maven-cli/src/main/java/org/apache/maven/cling/invoker/LookupInvoker.java

```diff
@@ -407,11 +407,7 @@ protected Consumer<String> doDetermineWriter(C context) {
         } else {
             // Given the terminal creation has been offloaded to a different thread,
             // do not pass directly the terminal writer
-            return msg -> {
-                PrintWriter pw = context.terminal.writer();
-                pw.println(msg);
-                pw.flush();
```

Comment:
might avoid this expensive operation, letting implementation check themself, when its time to trigger this.